utils/http.hh

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/shared_future.hh>
+#include <seastar/coroutine/all.hh>
+#include <seastar/http/client.hh>
+#include <seastar/net/dns.hh>
+#include <seastar/net/tls.hh>
+
+#include "seastarx.hh"
+#include "log.hh"
+
+namespace utils::http {
+
+class dns_connection_factory : public seastar::http::experimental::connection_factory {
```

Comment:
It took me many minutes to figure out what this class does. It's **not** a "dns connection factory", it doesn't connect to DNS, it's a "TCP/TLS connection factory with the ability to resolve domain names in the address". I think this class should be renamed, and if not, at least a comment added to explain its purpose.

Reply:
I extracted this code from the S3 client, as @xemul recommended.
I can add the comment but I would prefer not to change it beyond that.

Reply:
Ok